src/main/kotlin/org/openmbee/flexo/mms/Layer1Context.kt

```diff
@@ -385,7 +385,7 @@ class Layer1Context<TRequestContext: GenericRequest, out TResponseContext: Gener
 
         // compose output etag
         var outputEtag = if(bindings.size == 1) {
-            bindings[0]["__mms_etag"]!!.jsonObject["value"]!!.jsonPrimitive.content
+            bindings[0]["__mms_etag"]?.let { it.jsonObject["value"]!!.jsonPrimitive.content }?: ""
```

Comment:
in the else block it treats etag as non nullable, prob should do the same in else

Reply:
method is now obsolete. deleted